### PR TITLE
Wix rich content editor upload files@0.0.1

### PR DESCRIPTION
**Pull request by FerasAMasood**

### CODE REVIEW

File: examples/editor/src/PluginsConfig.js [hunk 1]
```diff
@@ -207,5 +208,30 @@ export default {
     //     desktop: () => InlineToolbarDecoration
     //   })
     // }
-  ]
+  ],
+  [UPLOAD_FILE_TYPE]: {
+    accept: '*',
```
Comment (bchmn) [inline examples/editor/src/PluginsConfig.js]: add a `onFilesChange: (files, updateEntity)` which is called when there is no `handleFileSelection` in the plugin settings

File: packages/common/src/Base/createBaseInsertPluginButton.jsx [hunk 2]
```diff
@@ -157,12 +162,12 @@ export default ({ blockType, button, helpers, pubsub, t }) => {
           dataHook={`${button.name}_file_input`}
           className={classNames(styles.button, styles.fileUploadButton)}
           onChange={this.handleFileChange}
-          accept="image/*"
-          multiple={button.multi}
+          multiple={button.multiple}
```
Comment (bchmn) [inline packages/common/src/Base/createBaseInsertPluginButton.jsx]: `button.multi`

File: packages/plugin-upload-file/src/toolbar/inline-upload-file-buttons.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    { keyName: 'sizeSmallLeft', type: BUTTONS.SIZE_SMALL_LEFT, mobile: false },
+    { keyName: 'sizeSmallCenter', type: BUTTONS.SIZE_SMALL_CENTER, mobile: false },
+    { keyName: 'sizeSimallRight', type: BUTTONS.SIZE_SMALL_RIGHT, mobile: false },
+    { keyName: 'separator2', type: BUTTONS.SEPARATOR, mobile: false },
+    {
+      keyName: 'replace',
+      type: BUTTONS.FILES,
+      icon: MediaReplaceIcon,
+      settings,
+      tooltipTextKey: t('replace_file_tooltip'),
```
Comment (bchmn) [inline packages/plugin-upload-file/src/toolbar/inline-upload-file-buttons.jsx]: `FileUpload_Replace_Tooltip`
+ add to `messages_en.json`

File: packages/plugin-upload-file/src/toolbar/insert-upload-file-buttons.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { DEFAULTS } from '../upload-file-component';
+import { TOOLBARS } from 'wix-rich-content-common';
+import { InsertPluginIcon } from '../icons';
+
+export default ({ helpers, t, settings }) => {
+  return [
+    {
+      type: 'file',
+      name: 'UploadFile',
+      tooltipText: t('upload_file_tooltip'),
```
Comment (bchmn) [inline packages/plugin-upload-file/src/toolbar/insert-upload-file-buttons.jsx]: `FileUpload_InsertButton_Tooltip`
+ add to `messages_en.json`

File: packages/plugin-upload-file/src/upload-file-component.jsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.styles = mergeStyles({ styles, theme: this.props.theme });
+  }
+
+  setPlayer = player => {
+    this.player = player;
+  };
```
Comment (bchmn) [inline packages/plugin-upload-file/src/upload-file-component.jsx]: remove

File: packages/plugin-upload-file/src/upload-file-component.jsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+UploadFileComponent.propTypes = {
+  componentData: PropTypes.object.isRequired,
+  componentState: PropTypes.object.isRequired,
+  store: PropTypes.object.isRequired,
+  blockProps: PropTypes.object.isRequired,
+  className: PropTypes.string.isRequired,
+  theme: PropTypes.object.isRequired,
+  t: PropTypes.func,
+};
```
Comment (bchmn) [inline packages/plugin-upload-file/src/upload-file-component.jsx]: remove everything except `componentData` and `theme`

File: packages/plugin-upload-file/src/upload-file-component.jsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+  setPlayer = player => {
+    this.player = player;
+  };
+
+  render() {
+    const { componentData } = this.props;
+    return (
+      <div data-hook="upload-file-component-container" className={styles.upload_file_container}>
+        <a href={componentData.fileURL} className={styles.upload_file_link}>
```
Comment (bchmn) [inline packages/plugin-upload-file/src/upload-file-component.jsx]: replace all of this with `<UploadFileViewer>`

File: packages/plugin-upload-file/src/upload-file-viewer.jsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  componentData: PropTypes.object.isRequired,
+  theme: PropTypes.object,
+  onReady: PropTypes.func,
+  onStart: PropTypes.func,
+  controls: PropTypes.bool,
+  width: PropTypes.string,
+  height: PropTypes.string,
+};
+
+UploadFileViewer.defaultProps = {
```
Comment (bchmn) [inline packages/plugin-upload-file/src/upload-file-viewer.jsx]: remove

File: packages/plugin-upload-file/src/upload-file-viewer.jsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+          <DocumentIcon className={styles.upload_file_icon} />
+          <span className={styles.upload_file_type}>{componentData.fileType}</span>
+        </div>
+        <span className={styles.upload_file_name}>{componentData.fileName}</span>
+      </a>
+    );
+  }
+}
+
+UploadFileViewer.propTypes = {
```
Comment (bchmn) [inline packages/plugin-upload-file/src/upload-file-viewer.jsx]: remove everything except componentData and theme

File: packages/plugin-upload-file/statics/data-schema.json [hunk 10]
```diff
@@ -0,0 +1,70 @@
+{
+  "$id": "http://path/to/schema.json",
+  "type": "object",
+  "properties": {
+    "src": {
```
Comment (bchmn) [inline packages/plugin-upload-file/statics/data-schema.json]: remove video properties + add your properties (with examples)

File: packages/plugin-upload-file/statics/styles/default-upload-file-styles.scss [hunk 11]
```diff
@@ -0,0 +1,2 @@
+@import "~wix-rich-content-common/dist/statics/styles/palette";
+@import "upload-file-viewer";
```
Comment (bchmn) [inline packages/plugin-upload-file/statics/styles/default-upload-file-styles.scss]: remove this import

File: packages/plugin-upload-file/statics/styles/upload-file-viewer.scss [hunk 12]
```diff
@@ -0,0 +1,56 @@
+@import '~wix-rich-content-common/dist/statics/styles/mixins';
+
+.upload_file_container {
```
Comment (bchmn) [inline packages/plugin-upload-file/statics/styles/upload-file-viewer.scss]: move this class to  `default-upload-file-styles.scss`

File: packages/plugin-upload-file/statics/styles/upload-file-viewer.scss [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+  box-sizing: border-box;
+}
+
+.upload_file_icon_container  {
+  position:relative;
+  display: inline-block;
+  height: 32px;
+  padding: 5px 0px;
+  margin-right: 24px;
+  float: left;
```
Comment (bchmn) [inline packages/plugin-upload-file/statics/styles/upload-file-viewer.scss]: float is irrelevant in `inline-block`, double check that you need this

File: packages/plugin-upload-file/statics/styles/upload-file-viewer.scss [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+.upload_file_name {
+  display: inline-block;
+  height: 32px;
+  text-align: left;
+  font-size: 14px;
+  line-height: 24px;
+  padding: 5px 0px;
+  float: left;
```
Comment (bchmn) [inline packages/plugin-upload-file/statics/styles/upload-file-viewer.scss]: float is irrelevant in `inline-block`, double check that you need this

File: packages/common/src/Base/baseToolbarButton.jsx [hunk 15]
```diff
@@ -172,19 +180,25 @@ class BaseToolbarButton extends React.Component {
   };
 
   renderFilesButton = (buttonClassNames, styles) => {
-    const { theme, isMobile, t, tooltipTextKey, tabIndex } = this.props;
+    const { theme, isMobile, t, tooltipTextKey, tabIndex, settings } = this.props;
     const tooltipText = t(tooltipTextKey);
```
Comment (bchmn) [inline packages/common/src/Base/baseToolbarButton.jsx]:
```suggestion
    const accept = settings || {};
    const tooltipText = t(tooltipTextKey);
```

File: packages/common/src/Base/baseToolbarButton.jsx [hunk 16]
```diff
@@ -172,19 +180,25 @@ class BaseToolbarButton extends React.Component {
...
+      handleChange = this.handleFileChange;
+    }
     const replaceButtonWrapperClassNames = classNames(styles.buttonWrapper);
     const filesButton = (
       <div className={replaceButtonWrapperClassNames}>
-        <FileInput
+{       <FileInput
           className={classNames(buttonClassNames)} theme={theme} tabIndex={tabIndex}
-          dataHook={this.getDataHook()} onChange={this.handleFileChange} accept="image/*" multiple={this.props.multiple}
+          dataHook={this.getDataHook()} onChange={handleChange} onClick={() =>{handleClick(this.handleExternalFileChanged)}}  multiple={this.props.multiple}
```
Comment (bchmn) [inline packages/common/src/Base/baseToolbarButton.jsx]:
```suggestion
          dataHook={this.getDataHook()} onChange={handleChange} onClick={() =>{handleClick(this.handleExternalFileChanged)}}  multiple={this.props.multiple} accept={accept}
```

File: .gitignore [hunk 17]
```diff
@@ -8,6 +8,7 @@ yarn.lock
 target
 maven
 stats.html
+wix-rich-content.iml
```
Comment (bchmn) [inline .gitignore]:
```suggestion
*.iml
```
and move it down to be under the `#Webstorm` section

File: packages/plugin-upload-file/statics/styles/upload-file-viewer.scss (hunk 13 above)
Comment (FerasAMasood) [reply packages/plugin-upload-file/statics/styles/upload-file-viewer.scss]: the styled element is a span, the idea was to change it from inline to inline-block to make the height property effective.

File: packages/plugin-upload-file/statics/styles/upload-file-viewer.scss (hunk 14 above)
Comment (FerasAMasood) [reply packages/plugin-upload-file/statics/styles/upload-file-viewer.scss]: the styled element is a span, the idea was to change it from inline to inline-block to make the height property effective.

**Pull request closed without merging**